web/html/src/components/package/PackageListActionScheduler.tsx

```diff
@@ -0,0 +1,256 @@
+import * as React from "react";
+
+import { ActionChain, ActionSchedule } from "components/action-schedule";
+import { AsyncButton, Button } from "components/buttons";
+import { ActionChainLink, ActionLink } from "components/links";
+import { Messages, MessageType, Utils as MessagesUtils } from "components/messages";
+import { InnerPanel } from "components/panels/InnerPanel";
+import { Column } from "components/table/Column";
+import { SearchField } from "components/table/SearchField";
+import { Table } from "components/table/Table";
+
+import { localizedMoment } from "utils";
+import { Utils } from "utils/functions";
+import Network from "utils/network";
+
+type Props = {
+  serverId: number;
+  selectionSet: string;
+  actionChains: Array<ActionChain>;
+  icon: string;
+  listDataAPI: string;
+  scheduleActionAPI: string;
+  actionType: string;
+  listTitle: string;
+  listSummary: string;
+  listEmptyText: string;
+  listActionLabel: string;
+  listColumns: React.ReactNode[];
+  confirmTitle: string;
+};
+
+type State = {
+  messages: Array<MessageType>;
+  selectedPackages: string[];
+  confirmAction: boolean;
+  earliest: moment.Moment;
+  actionChain?: ActionChain;
+};
+
+export class PackageListActionScheduler extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      messages: [],
+      selectedPackages: [],
+      confirmAction: false,
+      earliest: localizedMoment(),
+    };
+  }
+
+  confirmAction = () => {
+    this.setState({
+      confirmAction: true,
+      messages: [],
+    });
+  };
+
+  backToList = () => {
+    this.setState({
+      confirmAction: false,
+      messages: [],
+    });
+  };
+
+  scheduleAction = () => {
+    const requestBody = {
+      actionType: this.props.actionType,
+      earliest: this.state.earliest,
+      actionChain: this.state.actionChain ? this.state.actionChain.text : null,
+      selectedPackages: this.state.selectedPackages,
+    };
+
+    Network.post(this.props.scheduleActionAPI, requestBody)
+      .then((data) => {
+        // Notify the successful outcome
+        const msg = MessagesUtils.info(
+          this.state.actionChain ? (
+            <span>
+              {t("Action has been successfully added to the ")}
+              <ActionChainLink id={data}>{this.state.actionChain.text}</ActionChainLink>
+            </span>
+          ) : (
+            <span>
+              {t("The action has been ")}
```

Comment:
This type of split translation won't work well for languages where the order of the words is different. I don't remember if our translation wrapper supports stuff like `{0}` with DOM nodes though. @parlt91 do you happen to know if we have any way to solve this well?

Reply:
I see your point. I kind of "blindly" copied this from `highstate.tsx`, so in case there is also that occurrence to fix.

Reply:
The `t()` function supports dynamic parameters in the format `{n}`. I'm not sure if passing HTML in them is 100% safe, but it's one way to do it.

@mackdk you're right, we have many such mistakes accumulated over time, so we should at least fix them as we come across them (working on the same file, etc.).

*`t()` function implementation:*
https://github.com/uyuni-project/uyuni/blob/master/web/html/src/utils/translate.tsx#L34-L65

Reply:
From what I see,  JSX was supported only as main argument of the function `t()` like: 

```tsx
t(
    <span>Action has been successfully added to the <ActionChainLink id={data}>{this.state.actionChain.text}</ActionChainLink></span>
)
```

But I think this is just impossible to convert to a translation `msgid` for the pot file, isn't it? It wasn't in fact picked up by our translation scripts. 

JSX, on the other hand, wasn't working as one of the  `{n}` parameters, as in:

```tsx
{t("Action has been successfully added to the {0}",
    <ActionChainLink id={data}>{this.state.actionChain.text}</ActionChainLink>
)}
```

![image](https://user-images.githubusercontent.com/2292684/214575417-ede0ce18-8e41-4752-bc3f-befd1b9fb284.png)

So I added support that. @cbbayburt / @Etheryte please double check if the implementation is acceptable and safe.

However this still has issues. For example in this case the English text is `The action has been <>scheduled</>` and we want the react component to be wrapped around the word `scheduled`. But the part we want as link might differ depending on the language, and also the word "scheduled" may translate differently in the target language depending on the way it's used, so it's not fully correct to have it splitted like I'm doing now.

Reply:
Imo at this point it would be reasonable to toss `t()` and integrate something like [`react-intl`](https://formatjs.io/docs/react-intl/). `t()` is too simplistic for the use cases we want and other libraries have already solved all of these problems.